apps/src/music/views/UpdateTimer.tsx

```diff
@@ -0,0 +1,58 @@
+import {ProgressManagerContext} from '@cdo/apps/labs/progress/ProgressContainer';
+import React, {useCallback, useContext, useEffect, useRef} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+import MusicPlayer from '../player/MusicPlayer';
+import {MusicState, setCurrentPlayheadPosition} from '../redux/musicRedux';
+
+type UpdateTimerProps = Pick<MusicPlayer, 'getCurrentPlayheadPosition'> & {
+  updateHighlightedBlocks: () => void;
+};
+
+const UPDATE_RATE = 1000 / 30; // 30 times per second
+
+/**
+ * Utility component that updates various parts of the Music Lab interface
+ * (playhead position, highlighted blocks, progress state) at a specific
+ * rate while playback is in progress.
+ */
+const UpdateTimer: React.FunctionComponent<UpdateTimerProps> = ({
```

Comment:
are these moved here to allow us to use functional component logic?

Reply:
Pretty much, yeah - accessing context objects inside class components is a bit more cumbersome so I thought it would be better to pull this out of music view for now. Also just helps move some logic outside of MusicView